naturf/tests/test_nodes.py

```diff
@@ -105,6 +105,50 @@ class TestCase:
                 actual,
                 f"failed test {case.name} expected {expected}, actual {actual}"
             )
+    
+    def test_wall_angle_direction_length(self):
+    """Test that the function wall_angle_direction_length returns the correct angle, direction, and length."""
+    
+    polyext = [[0,1], [1,1], [1,0], [0,0], [0,1]]
+    polyint = [[0.25, 0.25], [0.25, 0.75], [0.75, 0.75], [0.75, 0.25]]
+    
+    @dataclass
+    class Testcase:
+        name: str
+        input: List[Polygon]
+        expected: List[int]
+    
+    testcases = [
+        Testcase(name="square", input=[Polygon(polyext)], expected=[pd.concat(pd.Series([0.0, -90.0, 180.0, 90.0], name="wall_angle"),
```

Comment:
should be `TestCase` and we should use `Settings.wall_angle` rather than `"wall_angle"` (same for `"wall_direction"` and `"wall_length"`)